server/planning/utils.py

```diff
@@ -1,10 +1,85 @@
-from typing import Union
+from typing import Union, List, Dict, Any
 from bson.objectid import ObjectId
 from bson.errors import InvalidId
+from datetime import timedelta, datetime
+from flask_babel import format_time, format_datetime, lazy_gettext
+from eve.utils import str_to_date
+import arrow
+from flask import current_app as app
+
+ContactDict = Dict[str, Union[str, List[str]]]
+FormattedContacts = List[ContactDict]
 
 
 def try_cast_object_id(value: str) -> Union[ObjectId, str]:
     try:
         return ObjectId(value)
     except InvalidId:
         return value
+
+
+def get_formatted_contacts(event: Dict[str, Any]) -> FormattedContacts:
+    contacts = event.get("event_contact_info", [])
+    formatted_contacts = []
+    for contact in contacts:
+        if contact.get("public", False):
+            formatted_contacts.append(
+                {
+                    "name": " ".join(
+                        [
+                            c
+                            for c in [
+                                contact.get("first_name"),
+                                contact.get("last_name"),
+                            ]
+                            if c
+                        ]
+                    ),
+                    "organisation": contact.get("organisation", ""),
+                    "email": contact.get("contact_email", []),
+                    "phone": [c.get("number") for c in contact.get("contact_phone", []) if c.get("public")],
+                    "mobile": [c.get("number") for c in contact.get("mobile", []) if c.get("public")],
+                    "website": contact.get("website", ""),
+                }
+            )
+    return formatted_contacts
+
+
+def parse_date(datetime: Union[str, datetime]) -> datetime:
+    """Return datetime instance for datetime."""
+    if isinstance(datetime, str):
+        try:
+            return str_to_date(datetime)
+        except ValueError:
+            return arrow.get(datetime).datetime
+    return datetime
+
+
+def time_short(datetime: datetime):
+    if datetime:
+        return format_time(parse_date(datetime), app.config.get("TIME_FORMAT_SHORT", "HH:mm"))
+
+
+def date_short(datetime: datetime):
+    if datetime:
+        return format_datetime(parse_date(datetime), app.config.get("DATE_FORMAT_SHORT", "dd/MM/yyyy"))
+
+
+def get_event_formatted_dates(event: Dict[str, Any]) -> str:
+    DAY_IN_MINUTES = 24 * 60 - 1
```

Comment:
what's this `-1`?

Reply:
the -1 is subtracting one minute to avoid counting the very last minute of the day. So instead of counting 
`24 hours * 60 minutes = 1440` minutes in a day, it's counting 1439 minutes to exclude the last minute. This could be to avoid overlapping with the start time of the next day or to maintain consistency in time calculations

Reply:
it might make events ending on `23:59:59` handled wrong, I would then expect it to work with seconds and not minutes